### PR TITLE
Add an advanced mode to vega / vega lite chart

### PR DESCRIPTION
**Pull request by AlasDiablo**
- [x] Vega Lite
  - [x] bar chart
  - [x] bubble plot
  - [x] cartography
  - [x] heatmap
  - [x] pie chart
- [x] Vega
  - [x] flow map
  - [x] radar chart
- [x] Admin panel
  - [x] Add a translation to the rest button 
  - [x] Use a correct color for the rest button
- [x] Other refactor
  - [x] add comment to not easily understandable function
  - [x] convert view component to function component
    - [x] bar chart
    - [x] bubble plot
    - [x] cartography
    - [x] heatmap
    - [x] pie chart 
    - [x] flow map
    - [x] radar chart

## Error message

Screenshot_20231002_100517

### CODE REVIEW

**Comment by AlasDiablo** [comment]
[Permettre l'édition de la configuration vega lite des graphiques standards](https://trello.com/c/63AQpGzP/385-permettre-l%C3%A9dition-de-la-configuration-vega-lite-des-graphiques-standards)

File: src/app/custom/translations.tsv [hunk 1]
```diff
@@ -956,3 +956,5 @@
 "r-sub-resources-co-occurrences_description" 	"Identifies sub-resources that co-occur with a given sub-resource and calculates the number of these co-occurrences." 	"Identifie les sous-ressources co-occurrentes avec une sous-ressource donnée et calcule le nombre de ces co-occurrences."
 "r-close-by_title" 	"close-by" 	"close-by"
 "r-close-by_description" 	"Is used to create the graph aster plot." 	"Est utilisée pour créer le graphique aster plot."
+regenerate_vega_lite_spec	Regenerate Vega Lite specifications	Régénérer la spécifications Vega Lite
```
Comment (parmentf) [inline src/app/custom/translations.tsv]:
```suggestion
regenerate_vega_lite_spec	Regenerate Vega Lite specifications	Régénérer la spécification Vega Lite
```

File: src/app/custom/translations.tsv [hunk 2]
```diff
@@ -956,3 +956,5 @@
...
+regenerate_vega_spec	Regenerate Vega specifications	Régénérer la spécifications Vega
```
Comment (parmentf) [inline src/app/custom/translations.tsv]:
```suggestion
regenerate_vega_spec	Regenerate Vega specifications	Régénérer la spécification Vega
```

**Review by touv — COMMENTED** [review]
note for later: add a specific test for each graph

File: src/app/js/formats/vega-lite/models/BarChart.js [hunk 3]
```diff
@@ -172,9 +175,10 @@ class BarChart extends BasicChart {
 
     /**
      * Function use for rebuild the edited spec
```
Comment (parmentf) [inline src/app/js/formats/vega-lite/models/BarChart.js]:
```suggestion
     * Rebuild the edited spec
```

File: src/app/js/formats/vega-lite/models/Cartography.js [hunk 4]
```diff
@@ -26,9 +28,9 @@ class Cartography extends BasicChart {
 
     /**
      * Function use for rebuild the edited spec
```
Comment (parmentf) [inline src/app/js/formats/vega-lite/models/Cartography.js]:
```suggestion
     * Rebuild the edited spec
```

File: src/app/js/formats/vega-lite/models/HeatMap.js [hunk 5]
```diff
@@ -112,9 +114,9 @@ class HeatMap extends BasicChart {
 
     /**
      * Function use for rebuild the edited spec
```
Comment (parmentf) [inline src/app/js/formats/vega-lite/models/HeatMap.js]:
```suggestion
     * Rebuild the edited spec
```

File: src/app/js/formats/vega-lite/models/PieChart.js [hunk 6]
```diff
@@ -20,9 +23,9 @@ class PieChart extends BasicChart {
 
     /**
      * Function use for rebuild the edited spec
```
Comment (parmentf) [inline src/app/js/formats/vega-lite/models/PieChart.js]:
```suggestion
     * Rebuild the edited spec
```

File: src/app/js/formats/vega/models/FlowMap.js [hunk 7]
```diff
@@ -1,15 +1,19 @@
 import { schemeBlues } from 'd3-scale-chromatic';
 import { VEGA_ACTIONS_WIDTH } from '../../vega-lite/component/vega-lite-component/VegaLiteComponent';
+import flowMapVG from './json/flow_map.vg.json';
+import deepClone from 'lodash.clonedeep';
+import BasicChartVG from './BasicChartVG';
 
 /**
  * Class use for create radar chart spec
```
Comment (parmentf) [inline src/app/js/formats/vega/models/FlowMap.js]: I thought it was a FlowMap...

**Review by parmentf — CHANGES_REQUESTED** [review]

File: cypress/integration/graph_page.spec.js [hunk 8]
```diff
@@ -6,6 +6,13 @@ import * as searchDrawer from '../support/searchDrawer';
 
 describe('Graph Page', () => {
     beforeEach(() => {
+        // ResizeObserver don't like when the app has to many renders / re-renders
+        // and throw an exception to say I wait for the next paint
```
Comment (parmentf) [inline cypress/integration/graph_page.spec.js]:
```suggestion
        // and throws an exception to say "I wait for the next paint"
```

File: cypress/integration/graph_page.spec.js [hunk 9]
```diff
@@ -6,6 +6,13 @@ import * as searchDrawer from '../support/searchDrawer';
 
 describe('Graph Page', () => {
     beforeEach(() => {
+        // ResizeObserver don't like when the app has to many renders / re-renders
```
Comment (parmentf) [inline cypress/integration/graph_page.spec.js]:
```suggestion
        // ResizeObserver doesn't like when the app has to many renders / re-renders
```

File: cypress/integration/model.spec.js [hunk 10]
```diff
@@ -7,6 +7,13 @@ import * as searchDrawer from '../support/searchDrawer';
 describe('Model Page', () => {
     describe('handling old models', () => {
         beforeEach(() => {
+            // ResizeObserver don't like when the app has to many renders / re-renders
```
Comment (parmentf) [inline cypress/integration/model.spec.js]:
```suggestion
            // ResizeObserver doesn't like when the app has to many renders / re-renders
```

File: cypress/integration/model.spec.js [hunk 11]
```diff
@@ -7,6 +7,13 @@ import * as searchDrawer from '../support/searchDrawer';
...
+            // and throw an exception to say I wait for the next paint
```
Comment (parmentf) [inline cypress/integration/model.spec.js]:
```suggestion
            // and throws an exception to say "I wait for the next paint"
```

File: src/app/js/formats/vega-lite/component/bar-chart/BarChartView.js [hunk 12]
```diff
@@ -16,69 +15,123 @@ import {
...
+        const specBuilder = new BarChart();
 
-        // Set all bar chart parameter the chosen by the administrator
-
-        barChartSpec.setAxisDirection(
-            this.props.direction === 'vertical'
-                ? AXIS_VERTICAL
-                : AXIS_HORIZONTAL,
+        specBuilder.setAxisDirection(
+            direction === 'vertical' ? AXIS_VERTICAL : AXIS_HORIZONTAL,
```
Comment (parmentf) [inline src/app/js/formats/vega-lite/component/bar-chart/BarChartView.js]: Why not use `lodexDirectionToIdDirection` from `charsUtils.js`?

By the way, is `lodexDirectionToIdDirection` used anywhere?

File: src/app/js/formats/chartsUtils.js [hunk 13]
```diff
@@ -73,3 +73,19 @@ export const lodexOrderToIdOrder = orderBy => {
...
+export const lodexDirectionToIdDirection = direction => {
+    switch (direction) {
+        case 'vertical':
+            return AXIS_VERTICAL;
+        case 'horizontal':
+            return AXIS_HORIZONTAL;
+        default:
+            return INVALID_VALUE;
+    }
+};
```
Comment (parmentf) [inline src/app/js/formats/chartsUtils.js:91]: Is `lodexDirectionToIdDirection` used anywhere?

If not, remove it.

File: src/app/js/formats/vega-lite/component/vega-lite-component/VegaLiteComponent.js [hunk 14]
```diff
@@ -3,12 +3,12 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { isAdmin } from '../../../../user';
-import deepClone from 'lodash.clonedeep';
 import {
     VEGA_LITE_DATA_INJECT_TYPE_A,
     VEGA_LITE_DATA_INJECT_TYPE_B,
     VEGA_LITE_DATA_INJECT_TYPE_C,
 } from '../../../chartsUtils';
+import deepClone from 'lodash.clonedeep';
```
Comment (parmentf) [inline src/app/js/formats/vega-lite/component/vega-lite-component/VegaLiteComponent.js]: Why did you move that line?
Don't you place package import before local imports?

File: src/app/js/formats/vega-lite/models/BarChart.js [hunk 15]
```diff
@@ -14,9 +14,12 @@ import {
 } from '../../chartsUtils';
 import { VEGA_ACTIONS_WIDTH } from '../component/vega-lite-component/VegaLiteComponent';
 import BasicChart from './BasicChart';
+import barChartVL from './json/bar_chart.vl.json';
+import barChartLabelsVL from './json/bar_chart_labels.vl.json';
+import deepClone from 'lodash.clonedeep';
 
 /**
- * Class use for create bar chart spec
+ * Class used to create bar chart spec
```
Comment (parmentf) [inline src/app/js/formats/vega-lite/models/BarChart.js]: Not sure this comment convey much information.

File: src/app/js/formats/vega-lite/models/Cartography.js [hunk 16]
```diff
@@ -1,17 +1,19 @@
 import BasicChart from './BasicChart';
 import { MAP_EUROPE, MAP_FRANCE, MAP_WORLD } from '../../chartsUtils';
 import { VEGA_ACTIONS_WIDTH } from '../component/vega-lite-component/VegaLiteComponent';
+import cartographyVL from './json/cartography.vl.json';
+import deepClone from 'lodash.clonedeep';
 
 /**
- * Class use for create cartography spec
+ * Class used to create cartography spec
```
Comment (parmentf) [inline src/app/js/formats/vega-lite/models/Cartography.js]: Useless comment?

File: src/app/js/formats/vega-lite/models/HeatMap.js [hunk 17]
```diff
@@ -1,16 +1,18 @@
 import BasicChart from './BasicChart';
 import { LABEL_ASC, LABEL_DESC } from '../../chartsUtils';
+import heatmapVL from './json/heatmap.vl.json';
+import deepClone from 'lodash.clonedeep';
 
 /**
- * Class use for create heatmap spec
+ * Class used to create heatmap spec
```
Comment (parmentf) [inline src/app/js/formats/vega-lite/models/HeatMap.js]: Is this really useful?

File: src/app/js/formats/vega-lite/models/PieChart.js [hunk 18]
```diff
@@ -1,16 +1,19 @@
 import BasicChart from './BasicChart';
+import pieChartVL from './json/pie_chart.vl.json';
+import pieChartLabelsVL from './json/pie_chart_labels.vl.json';
+import deepClone from 'lodash.clonedeep';
 
 /**
- * Class use for create pie chart spec
+ * Class used to create pie chart spec
```
Comment (parmentf) [inline src/app/js/formats/vega-lite/models/PieChart.js]: Yes, it is said on the next code line.

File: src/app/js/formats/vega/models/FlowMap.js [hunk 19]
```diff
@@ -1,15 +1,19 @@
 import { schemeBlues } from 'd3-scale-chromatic';
 import { VEGA_ACTIONS_WIDTH } from '../../vega-lite/component/vega-lite-component/VegaLiteComponent';
+import flowMapVG from './json/flow_map.vg.json';
+import deepClone from 'lodash.clonedeep';
+import BasicChartVG from './BasicChartVG';
 
 /**
- * Class use for create radar chart spec
+ * Class use for create "flow map" chart spec
```
Comment (parmentf) [inline src/app/js/formats/vega/models/FlowMap.js]: Really ? ;)

File: src/app/js/formats/vega/models/RadarChart.js [hunk 20]
```diff
@@ -1,15 +1,19 @@
 import { MONOCHROMATIC_DEFAULT_COLORSET } from '../../colorUtils';
 import { SCALE_LINEAR } from '../../chartsUtils';
+import radarChartVG from './json/radar_chart.vg.json';
+import deepClone from 'lodash.clonedeep';
+import BasicChartVG from './BasicChartVG';
 
 /**
  * Class use for create radar chart spec
```
Comment (parmentf) [inline src/app/js/formats/vega/models/RadarChart.js]: Oh!

**Review by parmentf — CHANGES_REQUESTED** [review]

File: src/app/js/formats/vega-lite/component/vega-lite-component/VegaLiteComponent.js (hunk 14 above)
Comment (AlasDiablo) [reply src/app/js/formats/vega-lite/component/vega-lite-component/VegaLiteComponent.js]: auto import, put it at the end...

File: src/app/js/formats/chartsUtils.js (hunk 13 above)
Comment (AlasDiablo) [reply src/app/js/formats/chartsUtils.js:91]: `lodexDirectionToIdDirection` is use in `BarChartAdmin`, i will also use it in `BarChartView`

File: src/app/js/formats/vega/models/RadarChart.js (hunk 20 above)
Comment (AlasDiablo) [reply src/app/js/formats/vega/models/RadarChart.js]: ?

File: src/app/js/formats/vega-lite/models/Cartography.js (hunk 16 above)
Comment (parmentf) [reply src/app/js/formats/vega-lite/models/Cartography.js]: As the next line say `class Cartography`... the comment does not say anything more (except if "spec" is something needing to be explicit?).

File: src/app/js/formats/vega-lite/models/HeatMap.js (hunk 17 above)
Comment (parmentf) [reply src/app/js/formats/vega-lite/models/HeatMap.js]: Next line of code says `class HeatMap`.

File: src/app/js/formats/vega/models/RadarChart.js (hunk 20 above)
Comment (parmentf) [reply src/app/js/formats/vega/models/RadarChart.js]: The next line says `class RadarChart`, which contains the same information.

Is this comment really informative?

**Comment by touv** [comment]
si les modifications de François sont prises en compte peut-être que cette PR pourrait être mergée ...

Sinon, je pense qu'il aurait été plus simple à relire de faire une PR par graphique, même si il s'agit du même type de modification. à voir ce qu'en pense @parmentf

**Review by touv — APPROVED** [review]

**Review by touv — APPROVED** [review]

**Review by parmentf — APPROVED** [review]
